game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/BattleCalculatorPanel.java

```diff
@@ -1062,185 +1061,27 @@
     revalidate();
   }
 
-  /**
-   * Set initial attacker and defender.
-   *
-   * <p>Please read the source code for the order of the players and conditions involved.
-   */
   private void setupAttackerAndDefender() {
+    final Tuple<Optional<GamePlayer>, Optional<GamePlayer>> t =
```

Comment:
Replace `Tuple` with public inner class `AttackerAndDefender` (of `OpponentSelector`) with public variables `attacker` and `defender`.